app/src/main/java/com/example/currencyconverter/ui/di/ServiceLocator.kt

```diff
@@ -14,18 +17,37 @@ object ServiceLocator {
         NetworkService()
     }
 
+    private val domainMapper by lazy {
+        CurrencyDomainMapper()
+    }
+
+    private val networkMapper by lazy {
+        CurrencyNetworkMapper()
+    }
+
+    private val databaseMapper by lazy {
+        CurrencyDatabaseMapper()
+    }
+
+
+    private var valueInputValidator : ValueInputValidator? = null
+
     @Volatile
-    var currencyRepository: CurrencyRepository? = null
+    private var currencyRepository: CurrencyRepository? = null
 
     fun provideCurrencyRepository(context: Context): CurrencyRepository {
-        synchronized(this) {
             return currencyRepository ?: createCurrencyRepository(context)
```

Comment:
Seems you forgot to set currencyRepository with the newly created repository. So every time you call this function the new repository will be created.